src/telliot_feeds/reporters/tips/multicall_functions/multicall_autopay.py

```diff
@@ -25,67 +24,33 @@ async def timestamp_index_and_current_values_call(
-
-        if not databefore_resp:
-            return None, error_status("No response returned from getDataBefore batch multicall")
-
-        for feed in feeds:
-            # set QueryIdandFeedDetails attributes
-            feed.current_value_timestamp = databefore_resp[("current_value_timestamp", feed.query_id)]
-            feed.current_queryid_value = databefore_resp[("current_value", feed.query_id)]
-            feed.current_value_index = databefore_resp[("current_value_index", feed.query_id)]
-            feed.month_old_index = databefore_resp[("month_old_index", feed.query_id)]
-
-        return feeds, status
-
-    async def timestamps_list_call(
-        self, feeds: list[QueryIdandFeedDetails]
-    ) -> tuple[Optional[list[QueryIdandFeedDetails]], ResponseStatus]:
-        """Batch call getTimestampbyQueryIdandIndex to fetch a list of timestamps for each query id
-
-        Args:
-        - feeds: a list of QueryIdandFeedDetails object
-
-        Return: a list of QueryIdandFeedDetails object that carries query id and its month long timestamps list
-        """
         calls = [
-            self.get_timestamp_by_query_id_and_index(to_bytes(hexstr=feed.query_id), index=idx)
+            self.get_multiple_values_before(
+                query_id=feed.query_id, now_timestamp=now_timestamp, max_age=max_age, max_count=max_count
+            )
             for feed in feeds
-            for idx in range(feed.month_old_index, feed.current_value_index)
         ]
-
         if not len(calls):
-            return None, error_status("No getTimestampbyQueryIdandIndex Calls to assemble")
+            return None, error_status("Unable to assemble getMultipleValues Call object")
 
-        timestampby_queryid_index_resp, status = await self.multi_call(calls)
+        multiple_values_response, status = await self.multi_call(calls)
 
         if not status.ok:
             return None, status
 
-        if not timestampby_queryid_index_resp:
-            return None, error_status("No response returned from getTimestampbyQueryIdandIndex batch multicall")
-
-        # converts from: {(current_value, queryid): 0, (current_timestamp, queryid): 123}
-        # to: {queryid: [0, 123]}
-        clean_resp = filter_batch_result(timestampby_queryid_index_resp)
-
         for feed in feeds:
-            feed.queryId_timestamps_list = clean_resp[feed.query_id]
+            values = multiple_values_response[("values_array", feed.query_id)]
+            timestamps = multiple_values_response[("timestamps_array", feed.query_id)]
+            # short circuit the loop since None means failed response and can't calculate tip accurately
+            if values is None:
+                note = "getMultipleValuesBefore call failed"
+                return None, error_status(note)
+            feed.current_value_timestamp = timestamps[-1] if timestamps else 0
+            feed.current_queryid_value = values[-1] if values else b""
+
+            values_lis = values[:-1]
```

Comment:
will this throw index error if values len is 0?

Reply:
line 51? no, it'll just return an empty list if len is 0.